src/js/subtraction/components/Selector.js

```diff
@@ -1,27 +1,91 @@
-    ));
-    if (!subtractions.length) {
+export const SampleInputContainer = styled.div`
+    position: relative;
+    display: flex;
+`;
+
+export const IconContainer = styled.div`
+    position: absolute;
+    margin-left: 92%;
+    margin-top: 10px;
+`;
+
+export const WorkflowItem = styled(BoxGroupSection)`
+    display: flex;
+    flex-direction: row;
+    border: 1px solid;
+    border-color: ${props => props.theme.color.greyLight};
+`;
+
+export const SampleSubtractionSelector = ({
+    name,
+    noun,
+    selected,
+    subtractions,
+    onChange,
+    requireCheckbox,
+    requireInput,
+    term,
+    setTerm,
+    subtractionOptions
+}) => {
+    const subtractionComponents = subtractions.map(item => {
+        const result = item.id ? item : item.item;
         return (
-            <NoneFoundBox noun="subtractions">
-                <Link to="/subtractions"> Create one</Link>.
-            </NoneFoundBox>
+            <MultiSelectorItem
+                key={result.id}
+                name={result.name}
+                value={result.id}
+                id={result.id}
+                requireCheckbox={requireCheckbox}
+            >
+                <span>{result.name}</span>
+            </MultiSelectorItem>
         );
-    }
+    });
+
     return (
-        <MultiSelector name={name} noun={noun} selected={selected} onChange={onChange}>
-            {subtractionComponents}
-        </MultiSelector>
+        <>
+            {requireInput === true && (
+                <>
+                    {subtractionOptions.length - selected.length > 0 && (
+                        <SampleInputContainer>
+                            <Input
+                                placeholder="Filter subtractions"
+                                aria-label="Filter items"
+                                autoFocus
+                                value={term}
+                                onChange={e => setTerm(e.target.value)}
+                            />
+                            <IconContainer>
+                                <Icon name="times fa-fw" onClick={() => setTerm("")} aria-label="clear" />
+                            </IconContainer>
+                        </SampleInputContainer>
+                    )}
+                    {subtractionOptions.length === selected.length && subtractionOptions.length > 0 && (
+                        <WorkflowItem>All subtractions selected</WorkflowItem>
+                    )}
+                </>
+            )}
+            {subtractionOptions.length - selected.length > 0 && (
+                <MultiSelector name={name} noun={noun} selected={selected} onChange={onChange}>
+                    {subtractionComponents}
+                </MultiSelector>
+            )}
+
+            {!subtractionOptions.length && (
+                <NoneFoundBox noun="subtractions">
+                    <Link to="/subtractions"> Create one</Link>.
+                </NoneFoundBox>
+            )}
```

Comment:
Since this being true precludes all other states, I would just have this as a separate return statement the triggers if there are no subtraction options.